### PR TITLE
Support for loading Material list from StringVector

### PR DESCRIPTION
**Pull request by tijon1**
Makes life a lot easier when dealing with material lists.
Makes current file Material List also use the new function.

### CODE REVIEW

File: Source/Urho3D/Graphics/StaticModel.cpp [hunk 1]
```diff
@@ -317,6 +317,13 @@ void StaticModel::ApplyMaterialList(const ea::string& fileName)
         return;
 
     const StringVector lines = file->ReadLines();
+    ApplyMaterialList(lines);
+}
+
+void StaticModel::ApplyMaterialList(const StringVector& lines)
```
Comment (gleblebedev) [inline Source/Urho3D/Graphics/StaticModel.cpp:323]: Why not to call 
    /// Set materials attribute.
    void SetMaterialsAttr(const ResourceRefList& value);
It takes ResourceRefList which is list of strings + resource type.

Comment (tijon1) [reply Source/Urho3D/Graphics/StaticModel.cpp:323]: I could do that, but I don't think a ResourceRefList will work with a StringVector without doing some converting. StringVector is what material lists are represented as, ModelView::ExportMaterialList is a example.

Comment (gleblebedev) [reply Source/Urho3D/Graphics/StaticModel.cpp:323]: I would say that ExportMaterialList  should return ResourceRefList then. @eugeneko @rokups thoughts, suggestions?

Comment (rokups) [reply Source/Urho3D/Graphics/StaticModel.cpp:323]: I am inclined to agree. For public API `ResourceRefList` would make more sense.. `ApplyMaterialList(const StringVector& lines)` looks more like a helper function whose place is in protected/private section.

Comment (tijon1) [reply Source/Urho3D/Graphics/StaticModel.cpp:323]: ExportMaterialList needs to return ResourceRefList then. I agree with such a change but ExportMaterialList needs to be changed too then.

**Comment by tijon1** [comment]
see https://github.com/rbfx/rbfx/pull/579

**Pull request closed without merging**